File: Quick_Sort.py
import random
def quick_sort(nums):
    return _sort(nums, 0, len(nums) - 1)

def _sort(nums, left, right):
    if left < right:
        reference = nums[left]
        l = left
        r = right
        while l < r:
            while nums[l] <= reference and l < right:
                l += 1
            while nums[r] >  reference and left < r:
                r -=1
            if l < r:
                nums[r], nums[l] = nums[l], nums[r]

        nums[left], nums[r] = nums[r], nums[left]

        _sort(nums, left,   r-1)
        _sort(nums,  r+1, right)
        return nums

def main():
    unsorted_list = [random.randint(0, 100) for _ in range(10)]
    print("Before Selection Sort\t: ", unsorted_list)
    quick_sort(unsorted_list)
    print("After Selevtion Sort\t: ", quick_sort(unsorted_list))

if __name__ == '__main__':
    main()

File: quickSort.cpp
#include<iostream>
#include<vector>

using namespace std;

void quickSort(vector<int> &nums, int start, int end){
    if(start>=end) return;
    int pivot = nums[start];
    int left = start+1;
    int right = end;
    while(1){
        while(left<end && nums[left]<=pivot)left++;
        while(start<right && nums[right]>=pivot)right--;
        if(left>=right)break;
        swap(nums[left], nums[right]);
    }
    swap(nums[start], nums[right]);
    quickSort(nums, start, right-1);
    quickSort(nums, right+1, end);
}

void swap(int &a, int &b){
    int tmp;
    tmp = a;
    a = b;
    b = tmp;
}

int main(){
    vector<int> nums = {6, 5, 1, 3, 4, 9, 8, 7, 2, 0, 6, 5, 10, 99, -20, 34};
    quickSort(nums, 0, nums.size()-1);
    for(auto num: nums)
        cout<<num<<' ';
    cout<<endl;
    return 0;
}